### PR TITLE
Add `gen jlcpcb <jlcpcbPartNumberOrUrl>` command

### PR DESCRIPTION
**Pull request by andrii-balitskyi**
wrt https://github.com/tscircuit/cli/issues/80

https://github.com/user-attachments/assets/4bb341d5-6c8d-46be-ac57-4b64d80f0156

### CODE REVIEW

File: cli/lib/cmd-fns/gen-jlcpcb-component.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import path from "path"
+import fs from "fs/promises"
+import { spawn } from "bun"
+import { AppContext } from "../util/app-context"
```
Comment (seveibar) [inline cli/lib/cmd-fns/gen-jlcpcb-component.ts:3]:
```suggestion
import { AppContext } from "../util/app-context"
import { /* ... */ } from "easyeda"
```

File: cli/lib/get-program.ts [hunk 2]
```diff
@@ -67,6 +67,14 @@ export const getProgram = (ctx: AppContext) => {
     .option("-t, --type <type>", "Output file type (png or svg)")
     .action((args) => CMDFN.render(ctx, args))
 
+  const genCmd = cmd.command("gen").description("Generate components")
+  genCmd
+    .command("jlcpcb <jlcpcbPartNumber>")
```
Comment (seveibar) [inline cli/lib/get-program.ts]: should eventually take a url (we can create issue for it if difficult)

File: cli/lib/cmd-fns/gen-jlcpcb-component.ts (hunk 1 above)
Comment (andrii-balitskyi) [reply cli/lib/cmd-fns/gen-jlcpcb-component.ts:3]: 46721b18d7857037b586d58e48eddb26c0ed042d

File: cli/lib/get-program.ts (hunk 2 above)
Comment (andrii-balitskyi) [reply cli/lib/get-program.ts]: ed418ce4246b63034eea40e8139a95552958de9c

**Review by seveibar — APPROVED** [review]

**Pull request merged**